index.js

```diff
@@ -2,10 +2,14 @@ const config = require('./lib/config');
 const tracer = require('./lib/tracer');
 const httpWrapper = require('./lib/http_wrapper');
 
-module.exports = overrides => {
+const init = overrides => {
   const configuration = config.load(overrides);
   httpWrapper.wrapHttp(configuration);
   httpWrapper.wrapHttps(configuration);
 
   tracer.enable();
 };
+
+const getHeader = name => tracer.currentTrace.context.get(name);
```

Comment:
Hey @jodysalt.

Could you explain what's the purpose of the `getHeader` method please? Where do you see it being used?

The headers can be accessed from the incoming http request in the app, so I don't think you'd need to expose it here.

Reply:
One of the good neat things about using the async hooks, is that you don't explicitly need to pass context around.

This means you could have a function like `hasFeature('feature-flag')` that can be safely accessed anywhere in the code this means you don't need to explicitly pass it through a chain of functions to make use of it (which massively simplifies things).

I would see `hasFeature` implemented something like the following (and sitting in ww-utils):

```javascript
const { getHeader } = require('hpropagate');

const hasFeature = (featureFlag) => {
  const featureFlagsHeader = getHeader('x-feature-flag') || '';
  const featureFlags = featureFlagsHeader.split(/\*,\*/);
  return featureFlags.include(featureFlag);
};
```

If you consider feature flags are normally short lived things, it means you can pass context around without having to alter much of the existing code. The alternative would be to pass through the feature flag as some sort of argument from the initial request to where ever you want to use it.

I could now simply use much function **anywhere** like the following:

```javascript
if(hasFeature('chain-of-responsibility')){
  // do this
}
```

Reply:
Makes sense! Thanks @jodysalt for the awesome explanation.